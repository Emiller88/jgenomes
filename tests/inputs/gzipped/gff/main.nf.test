nextflow_pipeline {

    name "gzipped gff"
    script "../../../../main.nf"
    tag "input"
    tag "gff"

    test("Should run with a gzipped gff") {
        when {
            params {
                input = "${projectDir}/tests/inputs/gzipped/gff/input.yml"
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(path("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml").yaml).match("software_versions") },
                // { assert snapshot(
                //     // FIXME workflow.trace.tasks().size(),
                // ).match("output_files")
                // }
            )
        }
    }
}
