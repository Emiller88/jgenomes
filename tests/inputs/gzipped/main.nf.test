nextflow_pipeline {

    name "gzipped inputs"
    script "../../../main.nf"
    tag "input"

    test("Should run with a gzipped gtf") {
        tag "gtf"

        when {
            params {
                input = "${projectDir}/tests/inputs/gzipped/gzip_gtf.yml"
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(UTILS.removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml")).match("software_versions") },
                // { assert snapshot(
                //     // FIXME workflow.trace.tasks().size(),
                // ).match("output_files")
                // }
            )
        }
    }
}
