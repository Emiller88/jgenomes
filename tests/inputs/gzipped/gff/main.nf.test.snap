{
    "software_versions": {
        "content": [
            {
                "BOWTIE2_BUILD": {
                    "bowtie2": "2.5.2"
                },
                "BWAMEM1_INDEX": {
                    "bwa": "0.7.17-r1188"
                },
                "BWAMEM2_INDEX": {
                    "bwamem2": "2.2.1"
                },
                "DRAGMAP_HASHTABLE": {
                    "dragmap": "1.2.1"
                },
                "GATK4_CREATESEQUENCEDICTIONARY": {
                    "gatk4": "4.5.0.0"
                },
                "MSISENSORPRO_SCAN": {
                    "msisensor-pro": "1.2.0"
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev",
                    "Nextflow": "24.04.4"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-12T11:53:26.907185887"
    }
}